src/Akka.Hosting/AkkaConfigurationBuilder.cs

```diff
@@ -84,6 +86,9 @@ public sealed class AkkaConfigurationBuilder
         internal readonly string ActorSystemName;
         internal readonly IServiceCollection ServiceCollection;
         internal readonly HashSet<SerializerRegistration> Serializers = new();
+        
+        // we use a name / registration dictionary to make health check registrations unique by name
+        internal readonly Dictionary<string, AkkaHealthCheckRegistration> HealthChecks = new();
```

Comment:
Reorganized the internal storage of the health check system to use a dictionary and make the name of the health check the primary key - this is to help ensure that there can't be duplicate entries of the same health checks using our built-in methods or any user-defined ones.

Reply:
Since equality is done by reference on the `AkkaHealthCheckRegistration` instances